test/TestApps/AppLifecycleTestApp/SingleInstanceTest.cpp

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#include "pch.h"
+#include "Helpers.h"
+#include <TestDef.h>
+
+using namespace winrt;
+using namespace winrt::Windows::System;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::ApplicationModel::Activation;
+using namespace winrt::Microsoft::Windows::AppLifecycle;
+using namespace std::chrono;
+
+wil::unique_event g_activated;
+
+IAsyncAction WaitForActivations()
+{
+    if (!(co_await resume_on_signal(g_activated.get(), 3000s)))
+    {
+        // We timed out waiting for our redirection.
+        THROW_WIN32(ERROR_TIMEOUT);
+    }
+}
+
+void OnActivated(const winrt::Windows::Foundation::IInspectable&, const AppActivationArguments& args)
+{
+    auto launchArgs = args.Data().as<IProtocolActivatedEventArgs>();
+    SignalPhase(c_testInstanceRedirectedPhaseEventName);
+    g_activated.SetEvent();
+}
+
+bool SingleInstanceTestSucceeded(const AppActivationArguments& args)
+{
+    g_activated.create();
+
+    auto keyInstance = AppInstance::FindOrRegisterForKey(c_testInstanceRedirectedPhaseEventName);
+    if (!keyInstance)
+    {
+        return false;
+    }
+
+    if (keyInstance.IsCurrent())
+    {
+        auto token = keyInstance.Activated(auto_revoke, [&keyInstance](const auto& sender, const AppActivationArguments& args) { OnActivated(sender, args); });
+
+        // Launch a second instance and test that it redirects to us.
```

Comment:
I wasn't sure if the MAX_INSTANCE_COUNT in the shared process list was trying to govern the number of simultaneous instances you could have redirect to the "one true instance".  Do you think there is any value in stressing this a little bit by launching a few more than 1 in rapid succession to see if the system holds up under a moderate amount of multi-process contention?

Reply:
Good idea to add a test that activates max+1 and verifies it fails as expected